pyproject.toml

```diff
@@ -253,6 +255,11 @@ ignore = [
 "examples/*" = ["T20", "RET504", "E501"]
 "docs/*" = ["T20", "RET504", "E501"]
 "mlflow/*" = ["PT018"]
+# TODO: Consider enabling UP007 and UP045 in the following files.
+# Note: Switching to the new union syntax may introduce behavior changes.
+"mlflow/types/type_hints.py" = ["UP007", "UP045"]
+"tests/types/test_type_hints.py" = ["UP007", "UP045"]
+"tests/pyfunc/test_pyfunc_model_with_type_hints.py" = ["UP007", "UP045"]
```

Comment:
To ensure there is no behavior change, ignore `UP007` and `UP045` in tests for scheme inference with type hints.